lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+        )
+    }
+}
+
+#[allow(unsafe_code)]
+fn setup_camera_orthographic(dom:&JsValue, matrix:&Matrix4<f32>) {
+    // Views to WASM memory are only valid as long the backing buffer isn't
+    // resized. Check documentation of IntoFloat32ArrayView trait for more
+    // details.
+    unsafe {
+        let matrix_array = matrix.js_buffer_view();
+        js::setup_camera_orthographic(&dom, &matrix_array)
+    }
+}
+
+
+
+// =========================
+// === Css3dRendererData ===
+// =========================
+
+#[derive(Debug)]
+pub struct Css3dRendererData {
+    pub front_dom    : HtmlElement,
+    pub back_dom     : HtmlElement,
+    pub front_camera : HtmlElement,
+    pub back_camera  : HtmlElement
+}
+
+impl Css3dRendererData {
+    pub fn new
+    ( front_dom:HtmlElement
+    , back_dom:HtmlElement
+    , front_camera:HtmlElement
+    , back_camera:HtmlElement) -> Self {
+        Self {front_dom,back_dom,front_camera,back_camera}
+    }
+}
+
+impl Css3dRendererData {
+    fn set_dimensions(&self, dimensions:Vector2<f32>) {
+        let width  = format!("{}px", dimensions.x);
+        let height = format!("{}px", dimensions.y);
+        let doms   = vec![&self.front_dom,&self.back_dom,&self.front_camera,&self.back_camera];
+        for dom in doms {
+            dom.set_property_or_panic("width" , &width);
+            dom.set_property_or_panic("height", &height);
+        }
+    }
+}
+
+// ====================
+// === Css3dRenderer ===
+// ====================
+
+/// A renderer for `Css3dObject`s.
+#[derive(Clone,Debug)]
+pub struct Css3dRenderer {
+    container      : DomContainer,
+    data           : Rc<Css3dRendererData>,
+    logger         : Logger
+}
+
+impl Css3dRenderer {
+    /// Creates a Css3dRenderer inside an element.
+    pub fn from_element<L:Into<Logger>>(logger:L, element:HtmlElement) -> Result<Self> {
+        let logger                     = logger.into();
+        let container                  = DomContainer::from_element(element);
+        let front_dom    : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_dom     : HtmlElement = dyn_into(create_element("div")?)?;
+        let front_camera : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_camera  : HtmlElement = dyn_into(create_element("div")?)?;
+
+        front_dom.set_property_or_panic("position", "absolute");
+        front_dom.set_property_or_panic("top", "0px");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("width", "100%");
+        front_dom.set_property_or_panic("height", "100%");
+        front_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("position", "absolute");
+        back_dom.set_property_or_panic("top", "0px");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("overflow", "hidden");
```

Comment:
I knwo this is old code. We can keep panics hjere, but if you will find an easy way to just emit warnigns instead of panics, lets do it. If not, lets do it in new code and refactor it as time goes.

Reply:
1. I renamed `set_property_or_panic` to `set_style_or_panic`.
2. I created a `set_style_or_warn` function taking `&Logger` to `warn`, which can be used on places where `Logger` is present.